app/controllers/faculty_access_controller.rb

```diff
@@ -38,4 +40,9 @@ def instructor_has_selected_subject(key)
     params[:apply] && params[:apply][:subjects] && params[:apply][:subjects][key] == '1'
   end
 
+  def disallow_signup
+    # value doesn't really matter, if set at all, signup hidden
+    params[:no_signup] = 1
+  end
```

Comment:
Done because the `authenticate_user!` filter that runs after this does a redirect to the login page.  That before filter keeps this param around, and it will appear in the URL "/login?no_signup=1" so hopefully this isn't too bad, tho I agree it kind of appears on its own.